### ISSUE TITLE
General pathing issue with the outpost docking module.

### ISSUE THREAD

**Issue by Xeavas** [Regalis11/Barotrauma#4075]
![Pathing Bug](https://user-images.githubusercontent.com/72674994/95895162-d300f480-0dbc-11eb-81c4-1d2f02ae789c.jpg)

Description : 
Opening the side hatches flooding the docking module causes the hatch above to be sealed off, but there is still a clear path back into the submarine downwards. However, as shown in the picture above, they swam around and out into open waters without any protection and died, rather than try to close the door or attempt to swim back into the sub for safety.

Steps to reproduce : 
Start a new SP campaign and spawn or get a couple of bots to follow you into the docking module, open the door and watch them fumble about.

Additional info : found in v0.10.6.0 (Unstable) but I think it's a general pathing bug not related to this version only.

**Comment by itchyOwl** [comment]
I guess it's because there's no waypoints leading outside of the docking module: ![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/097c9b9a-28c2-4c30-9f33-dcdd9052d73d). I can take a look into this.

**Comment by itchyOwl** [comment]
Adding waypoints outside the sub seems to fix it: ![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/7d909d24-873b-4923-ae8e-0a486e6b88fd)

Adding more waypoints around the outpost modules would be good, but it requires that we disable these waypoints when they are blocked by other modules. So that would be a topic for another ticket.

**Comment by itchyOwl** [comment]
Implemented in Regalis11/Barotrauma-development@4bd80e5.

**Comment by Xeavas** [comment]
Still able to replicate bug, seems to happen the most often while both airlock doors are open, main observation is them trying to spam open the upperhatch but due to flooding its forced closed, and instead remain in said airlock rather than swim down. However on some instances they do manage to find safety, unsure if it was luck or a quirk.
1)
![pathfind3](https://user-images.githubusercontent.com/72674994/97781199-af171e80-1bc4-11eb-9008-2e144de20e8a.jpg)
2)
![pathbug1](https://user-images.githubusercontent.com/72674994/97781195-ade5f180-1bc4-11eb-8916-55a892d29e0c.jpg)
3)
![Pathfind2](https://user-images.githubusercontent.com/72674994/97781200-b0e0e200-1bc4-11eb-882d-6192d0dc92c8.jpg)

**Comment by Rokvach** [comment]
Moving back to todo

**Comment by Xeavas** [comment]
@itchyOwl I am not sure if this is a separate bug, but I'll post it here since its related to the outpost docking hatch module.

Ordering a bot from inside the sub to "wait" and the specified location (the pause with a circle to the right), results in them equipping the dive suit correctly, going out of the sub's airlock, but can't manage to path through the docking port module.
_(the bot had a scooter on it to make it more obvious its trying to swim)_
**[Sometimes they also get stuck by trying to swim through the corner of the dugong's airlock on the outside]**
![pathingthing](https://user-images.githubusercontent.com/72674994/98405987-20187380-20a8-11eb-89f1-414bc60432bd.jpg)

**Comment by itchyOwl** [comment]
Bot now know how to get back inside the sub through the gaps, which should fix the main issue of this ticket.

Implemented in Regalis11/Barotrauma-development@7112915.

The issue Xeavas described is different one. I'll create a separate ticket for it.

**Comment by Rokvach** [comment]
Tested, wasn't able to reproduce. Looks to be working correctly. Closing.
